pincer/objects/guild/guild.py

```diff
@@ -486,6 +496,260 @@ async def kick(self, member_id: int):
+            the `ROLE_ICONS` feature) |default| :data:`None`
+        unicode_emoji : Optional[:class:`str`]
+            The role's unicode emoji as a standard emoji (if the guild
+            has the `ROLE_ICONS` feature) |default| :data:`None`
+        mentionable : Optional[:class:`bool`]
+            Whether the role should be mentionable |default| :data:`None`
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.role.Role`
+            The edited role object.
+        """
+        ...
+
+    async def edit_role(
+        self,
+        id: Snowflake,
+        reason: Optional[str] = None,
+        **kwargs
+    ) -> Role:
+        return Role.from_dict(
+            construct_client_dict(
+                self._client,
+                await self._http.patch(
+                    f"guilds/{self.id}/roles/{id}",
+                    data=kwargs,
+                    headers={"X-Audit-Log-Reason": reason}
+                    if reason is not None
+                    else {},
+                ),
+            )
+        )
+
+    async def delete_role(self, id: Snowflake, reason: Optional[str] = None):
+        """|coro|
+        Deletes a role.
+        Requires the `MANAGE_ROLES` permission.
+        Returns `204 No Content` on success.
+
+        Parameters
+        ----------
+        id : :class:`~pincer.utils.snowflake.Snowflake`
+            The role ID
+        reason : Optional[:class:`str`]
+            The reason for deleting the role |default| :data:`None`
+        """
+        await self._http.delete(
+            f"guilds/{self.id}/roles/{id}",
+            headers={"X-Audit-Log-Reason": reason}
+            if reason is not None
+            else {},
+        )
+
+    async def get_bans(self) -> AsyncGenerator[Ban, None]:
+        """|coro|
+        Fetches all the bans in the guild.
+
+        Returns
+        -------
+        AsyncGenerator[:class:`~pincer.objects.guild.ban.Ban`, :data:`None`]
+            An async generator of Ban objects.
+        """
+        data = await self._http.get(f"guilds/{self.id}/bans")
+        for i in data:
+            yield Ban.from_dict(construct_client_dict(self._client, i))
+
+    async def get_ban(self, id: Snowflake) -> Ban:
+        """|coro|
+        Fetches a ban from the guild.
+        Returns `404 Not Found` if the user is not banned.
+
+        Parameters
+        ----------
+        id : :class:`~pincer.utils.snowflake.Snowflake`
+            The user ID
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.ban.Ban`
+            The Ban object.
+        """
+        return Ban.from_dict(
+            construct_client_dict(
+                self._client,
+                await self._http.get(f"guilds/{self.id}/bans/{id}"),
+            )
+        )
+
+    async def unban(self, id: Snowflake, reason: Optional[str] = None):
+        """|coro|
+        Unbans a user from the guild.
+        Returns `204 No Content` on success.
```

Comment:
remove this line. it can confuse people to believe that the function returns something